app/GioForegroundService.java

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+package org.gioui;
+import android.app.Notification;
+import android.app.Service;
+import android.app.Notification;
+import android.app.Notification.Builder;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.os.IBinder;
+
+public class GioForegroundService extends Service {
+	public static final String CHANNEL_ID = "GioForegroundService";
+	public static final String serviceName = "GioForeground";
+	public static final int notificationID = 42;
+
+	@Override public int onStartCommand (Intent intent, int flags, int startId) {
+		return START_NOT_STICKY;
+	}
+	@Override public IBinder onBind(Intent intent) {
+		return null;
+	}
+	@Override public void onCreate() {
+		super.onCreate();
+		this.createNotificationChannel();
+		this.startForeground();
+	}
+	public void startForeground() {
+		// create the Intent that will bring GioActivity to foreground
+		Context ctx = getApplicationContext();
+		Intent resultIntent = new Intent(ctx, GioActivity.class);
+		PendingIntent pending = PendingIntent.getActivity(ctx, notificationID, resultIntent, Intent.FLAG_ACTIVITY_CLEAR_TASK);
+		// create the Notification that Android will display on the tray for this service
+		Notification.Builder builder = new Notification.Builder(ctx, CHANNEL_ID)
+			.setContentTitle(CHANNEL_ID)
+			.setSmallIcon(getResources().getIdentifier("@mipmap/ic_launcher", "drawable", getPackageName()))
+			.setContentText(serviceName)
```

Comment:
Title and text are not very informative. Perhaps foreground.StartForeground should include them? It would be fine to do in a follow-up except that it's an API change.

Reply:
I'm happy to iterate here until the API looks good - rather than break it later.